### ISSUE TITLE
Build issue with master branch

### ISSUE THREAD

**Issue by psifidotos** [dangvd/ksmoothdock#57]
in master branch trying to build, I get the following errors:

```
[ 72%] Building CXX object CMakeFiles/config_dialog_test.dir/config_dialog_test.cc.o
/home/michail/Development/ksmoothdock/ksmoothdock/src/config_dialog_test.cc: In member function ‘bool ksmoothdock::ConfigDialogTest::compareDouble(double, double)’:
/home/michail/Development/ksmoothdock/ksmoothdock/src/config_dialog_test.cc:71:21: error: call of overloaded ‘abs(double)’ is ambiguous
     return abs(x - y) < kDelta;
                     ^
In file included from /usr/include/c++/6/cstdlib:75:0,
                 from /usr/include/c++/6/bits/stl_algo.h:59,
                 from /usr/include/c++/6/algorithm:62,
                 from /usr/include/qt5/QtCore/qglobal.h:94,
                 from /usr/include/qt5/QtGui/qicon.h:43,
                 from /usr/include/qt5/QtWidgets/qabstractbutton.h:43,
                 from /usr/include/qt5/QtWidgets/QAbstractButton:1,
                 from /home/michail/Development/ksmoothdock/ksmoothdock/src/config_dialog.h:22,
                 from /home/michail/Development/ksmoothdock/ksmoothdock/src/config_dialog_test.cc:19:
/usr/include/stdlib.h:751:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
In file included from /usr/include/c++/6/bits/stl_algo.h:59:0,
                 from /usr/include/c++/6/algorithm:62,
                 from /usr/include/qt5/QtCore/qglobal.h:94,
                 from /usr/include/qt5/QtGui/qicon.h:43,
                 from /usr/include/qt5/QtWidgets/qabstractbutton.h:43,
                 from /usr/include/qt5/QtWidgets/QAbstractButton:1,
                 from /home/michail/Development/ksmoothdock/ksmoothdock/src/config_dialog.h:22,
                 from /home/michail/Development/ksmoothdock/ksmoothdock/src/config_dialog_test.cc:19:
/usr/include/c++/6/cstdlib:185:3: note: candidate: __int128 std::abs(__int128)
   abs(__GLIBCXX_TYPE_INT_N_0 __x) { return __x >= 0 ? __x : -__x; }
   ^~~
/usr/include/c++/6/cstdlib:180:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/6/cstdlib:172:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
make[2]: *** [CMakeFiles/config_dialog_test.dir/build.make:63: CMakeFiles/config_dialog_test.dir/config_dialog_test.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:252: CMakeFiles/config_dialog_test.dir/all] Error 2
make: *** [Makefile:139: all] Error 2
```

**Comment by dangvd** [comment]
Hi Michail,

Thanks for letting me know about the bug. I'm not sure why the compiler in your case pulled abs() from <cstdlib> instead of from <cmath>, but I'll make sure it doesn't happen again.

Cheers,
Viet
P.S.: Cool stuffs you're working on!

**Comment by psifidotos** [comment]
Thanks!! 
You too !!! :-)

I wanted to review ksmoothdock and takes ideas for improvements in Latte :-)

I had used ksmoothdock in the past and i liked it a lot!!!

**Comment by dangvd** [comment]
Thanks! We can all learn from each other. Good luck with Latte Dock!

**Comment by psifidotos** [comment]
> Good luck with Latte Dock!
You too!!

> We can all learn from each other
Definitely!

I like a lot cooperating! In general I consider the kde community one big family! :)
Just to be clear in any case, Latte developement team is open if you wanna be part!!

**Comment by dangvd** [comment]
> I like a lot cooperating! In general I consider the kde community one big family! :)
Just to be clear in any case, Latte developement team is open if you wanna be part!!

Thanks for that!
I think KSmoothDock and Latte Dock have different goals though, for example:
- KSmoothDock aims to follow the traditional Linux model of desktop panel with app menu/launchers/pager/taskbar/system tray rather than MacOSX's way of task management
- KSmoothDock is not planning to integrate with Plasma widgets
so it's better to keep the projects separate.

However, it'd be nice to keep in touch and discuss general dock/desktop stuffs from time to time :)

**Comment by psifidotos** [comment]
> so it's better to keep the projects separate.

no prob

> However, it'd be nice to keep in touch and discuss general dock/desktop stuffs from time to time :)

of course!!

**Comment by dangvd** [comment]
Released in v5.6
